### PR TITLE
Add Bilingual Corpus of Arabic-English Parallel Tweets

### PR DESCRIPTION
**Pull request by sumanthd17**
Added Bilingual Corpus of Arabic-English Parallel Tweets. The link to the dataset can be found [here](https://alt.qcri.org/wp-content/uploads/2020/08/Bilingual-Corpus-of-Arabic-English-Parallel-Tweets.zip) and the paper can be found [here](https://www.aclweb.org/anthology/2020.bucc-1.3.pdf)

- [x] Followed the instructions in CONTRIBUTING.md
- [x] Ran the tests successfully
- [x] Created the dummy data

### CODE REVIEW

**Comment by abhishekkrthakur** [comment]
IMO, the problem with this dataset is that it is not really a text/nlp dataset. These are just collections of tweet ids. So, ultimately, one needs to crawl twitter to get the actual text.

**Comment by sumanthd17** [comment]
That's true.

File: datasets/tweets_ar_en_parallel/tweets_ar_en_parallel.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        self.data_url = data_url
+        self.citation = citation
+        self.url = url
+
+
+class ParallelTweets(datasets.GeneratorBasedBuilder):
+    BUILDER_CONFIGS = [
+        ParallelTweetsConfig(name=name, description=_DESCRIPTION, data_url=_DATA_URL, citation=_CITATION, url=_URL)
+        for name in ["parallelTweets", "accountList", "countryTopicAnnotation"]
+    ]
```
Comment (lhoestq) [inline datasets/tweets_ar_en_parallel/tweets_ar_en_parallel.py:61]:
```suggestion
    ]
    BUILDER_CONFIG_CLASS = ParallelTweetsConfig
```
we need that to specify that we're using a custom config

File: datasets/tweets_ar_en_parallel/tweets_ar_en_parallel.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+import os
+
+import pandas as pd
+
+import datasets
+
+
+_CITATION = """\
+    @inproceedings{Mubarak2020bilingualtweets,
```
Comment (lhoestq) [inline datasets/tweets_ar_en_parallel/tweets_ar_en_parallel.py:13]:
```suggestion
@inproceedings{Mubarak2020bilingualtweets,
```

File: datasets/tweets_ar_en_parallel/tweets_ar_en_parallel.py [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+            **kwrags: keyword arguments frowarded to super
+        """
+        super(ParallelTweetsConfig, self).__init__(version=datasets.Version("1.0.0", ""), **kwrags)
+        self.description = description
+        self.data_url = data_url
+        self.citation = citation
+        self.url = url
+
+
+class ParallelTweets(datasets.GeneratorBasedBuilder):
```
Comment (lhoestq) [inline datasets/tweets_ar_en_parallel/tweets_ar_en_parallel.py:57]:
```suggestion
class TweetsArEnParallel(datasets.GeneratorBasedBuilder):
```
this must be the camel case version of the dataset script name `tweets_ar_en_parallel`

File: datasets/tweets_ar_en_parallel/dataset_infos.json [hunk 4]
```diff
@@ -0,0 +1 @@
...ces/bilingual_corpus_of_parallel_tweets", "license": "", "features": {"account": {"dtype": "string", "id": null, "_type": "Value"}, "country": {"num_classes": 12, "names": ["QA", "BH", "AE", "OM", "SA", "PL", "JO", "IQ", "Other", "EG", "KW", "SY"], "names_file": null, "id": null, "_type": "ClassLabel"}, "topic": {"num_classes": 12, "names": ["Gov", "Culture", "Education", "Sports", "Travel", "Events", "Business", "Science", "Politics", "Health", "Governoment", "Media"], "names_file": null, "id": null, "_type": "ClassLabel"}}, "post_processed": null, "supervised_keys": null, "builder_name": "parallel_tweets", "config_name": "countryTopicAnnotation", "version": {"version_str": "1.0.0", "description": "", "major": 1, "minor": 0, "patch": 0}, "splits": {"test": {"name": "test", "num_bytes": 6036, "num_examples": 200, "dataset_name": "parallel_tweets"}}, "download_checksums": {"https://alt.qcri.org/wp-content/uploads/2020/08/Bilingual-Corpus-of-Arabic-English-Parallel-Tweets.zip": {"num_bytes": 2937626, "checksum": "a2a20772745825c2e0180699083517128519975d85ec30f451aa5450209996e4"}}, "download_size": 2937626, "post_processing_size": null, "dataset_size": 6036, "size_in_bytes": 2943662}}
```
Comment (lhoestq) [inline datasets/tweets_ar_en_parallel/dataset_infos.json:1]:
```suggestion
{"parallelTweets": {"description": "    Twitter users often post parallel tweets\u2014tweets that contain the same content but are\n    written in different languages. Parallel tweets can be an important resource for developing\n    machine translation (MT) systems among other natural language processing (NLP) tasks. This\n    resource is a result of a generic method for collecting parallel tweets. Using the method,\n    we compiled a bilingual corpus of English-Arabic parallel tweets and a list of Twitter accounts\n    who post English-Arabic tweets regularly. Additionally, we annotate a subset of Twitter accounts\n    with their countries of origin and topic of interest, which provides insights about the population\n    who post parallel tweets.\n", "citation": "    @inproceedings{Mubarak2020bilingualtweets,\ntitle={Constructing a Bilingual Corpus of Parallel Tweets},\nauthor={Mubarak, Hamdy and Hassan, Sabit and Abdelali, Ahmed},\nbooktitle={Proceedings of 13th Workshop on Building and Using Comparable Corpora (BUCC)},\naddress={Marseille, France},\nyear={2020}\n}\n", "homepage": "https://alt.qcri.org/resources/bilingual_corpus_of_parallel_tweets", "license": "", "features": {"ArabicTweetID": {"dtype": "int64", "id": null, "_type": "Value"}, "EnglishTweetID": {"dtype": "int64", "id": null, "_type": "Value"}}, "post_processed": null, "supervised_keys": null, "builder_name": "tweets_ar_en_parallel", "config_name": "parallelTweets", "version": {"version_str": "1.0.0", "description": "", "major": 1, "minor": 0, "patch": 0}, "splits": {"test": {"name": "test", "num_bytes": 2667296, "num_examples": 166706, "dataset_name": "tweets_ar_en_parallel"}}, "download_checksums": {"https://alt.qcri.org/wp-content/uploads/2020/08/Bilingual-Corpus-of-Arabic-English-Parallel-Tweets.zip": {"num_bytes": 2937626, "checksum": "a2a20772745825c2e0180699083517128519975d85ec30f451aa5450209996e4"}}, "download_size": 2937626, "post_processing_size": null, "dataset_size": 2667296, "size_in_bytes": 5604922}, "accountList": {"description": "    Twitter users often post parallel tweets\u2014tweets that contain the same content but are\n    written in different languages. Parallel tweets can be an important resource for developing\n    machine translation (MT) systems among other natural language processing (NLP) tasks. This\n    resource is a result of a generic method for collecting parallel tweets. Using the method,\n    we compiled a bilingual corpus of English-Arabic parallel tweets and a list of Twitter accounts\n    who post English-Arabic tweets regularly. Additionally, we annotate a subset of Twitter accounts\n    with their countries of origin and topic of interest, which provides insights about the population\n    who post parallel tweets.\n", "citation": "    @inproceedings{Mubarak2020bilingualtweets,\ntitle={Constructing a Bilingual Corpus of Parallel Tweets},\nauthor={Mubarak, Hamdy and Hassan, Sabit and Abdelali, Ahmed},\nbooktitle={Proceedings of 13th Workshop on Building and Using Comparable Corpora (BUCC)},\naddress={Marseille, France},\nyear={2020}\n}\n", "homepage": "https://alt.qcri.org/resources/bilingual_corpus_of_parallel_tweets", "license": "", "features": {"account": {"dtype": "string", "id": null, "_type": "Value"}}, "post_processed": null, "supervised_keys": null, "builder_name": "tweets_ar_en_parallel", "config_name": "accountList", "version": {"version_str": "1.0.0", "description": "", "major": 1, "minor": 0, "patch": 0}, "splits": {"test": {"name": "test", "num_bytes": 20108, "num_examples": 1389, "dataset_name": "tweets_ar_en_parallel"}}, "download_checksums": {"https://alt.qcri.org/wp-content/uploads/2020/08/Bilingual-Corpus-of-Arabic-English-Parallel-Tweets.zip": {"num_bytes": 2937626, "checksum": "a2a20772745825c2e0180699083517128519975d85ec30f451aa5450209996e4"}}, "download_size": 2937626, "post_processing_size": null, "dataset_size": 20108, "size_in_bytes": 2957734}, "countryTopicAnnotation": {"description": "    Twitter users often post parallel tweets\u2014tweets that contain the same content but are\n    written in different languages. Parallel tweets can be an important resource for developing\n    machine translation (MT) systems among other natural language processing (NLP) tasks. This\n    resource is a result of a generic method for collecting parallel tweets. Using the method,\n    we compiled a bilingual corpus of English-Arabic parallel tweets and a list of Twitter accounts\n    who post English-Arabic tweets regularly. Additionally, we annotate a subset of Twitter accounts\n    with their countries of origin and topic of interest, which provides insights about the population\n    who post parallel tweets.\n", "citation": "    @inproceedings{Mubarak2020bilingualtweets,\ntitle={Constructing a Bilingual Corpus of Parallel Tweets},\nauthor={Mubarak, Hamdy and Hassan, Sabit and Abdelali, Ahmed},\nbooktitle={Proceedings of 13th Workshop on Building and Using Comparable Corpora (BUCC)},\naddress={Marseille, France},\nyear={2020}\n}\n", "homepage": "https://alt.qcri.org/resources/bilingual_corpus_of_parallel_tweets", "license": "", "features": {"account": {"dtype": "string", "id": null, "_type": "Value"}, "country": {"num_classes": 12, "names": ["QA", "BH", "AE", "OM", "SA", "PL", "JO", "IQ", "Other", "EG", "KW", "SY"], "names_file": null, "id": null, "_type": "ClassLabel"}, "topic": {"num_classes": 12, "names": ["Gov", "Culture", "Education", "Sports", "Travel", "Events", "Business", "Science", "Politics", "Health", "Governoment", "Media"], "names_file": null, "id": null, "_type": "ClassLabel"}}, "post_processed": null, "supervised_keys": null, "builder_name": "tweets_ar_en_parallel", "config_name": "countryTopicAnnotation", "version": {"version_str": "1.0.0", "description": "", "major": 1, "minor": 0, "patch": 0}, "splits": {"test": {"name": "test", "num_bytes": 6036, "num_examples": 200, "dataset_name": "tweets_ar_en_parallel"}}, "download_checksums": {"https://alt.qcri.org/wp-content/uploads/2020/08/Bilingual-Corpus-of-Arabic-English-Parallel-Tweets.zip": {"num_bytes": 2937626, "checksum": "a2a20772745825c2e0180699083517128519975d85ec30f451aa5450209996e4"}}, "download_size": 2937626, "post_processing_size": null, "dataset_size": 6036, "size_in_bytes": 2943662}}
```

`parallel_tweets` -> `tweets_ar_en_parallel`

**Review by lhoestq — APPROVED** [review]
Thanks for adding this one !

**Comment by lhoestq** [comment]
at least it's clear in the description that one needs to collect the tweets : 
```
This resource is a result of a generic method for collecting parallel tweets.
```

**Comment by sumanthd17** [comment]
Looks like this is failing for other datasets. Should I rebase it and push again?
Also rebasing and pushing is reflecting changes in many other files (ultimately forcing me to open a new branch and a new PR) any way to avoid this?

**Comment by lhoestq** [comment]
No let me merge this one directly

**Comment by lhoestq** [comment]
merging since the CI is fixed on master

**Pull request merged**